packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2546,20 +2595,35 @@ export function checkForWrongSuspensePriorityInDEV(sourceFiber: Fiber) {
 
 function flushSuspensePriorityWarningInDEV() {
   if (__DEV__) {
-    if (componentsWithSuspendedDiscreteUpdates !== null) {
+    if (
+      componentsWithSuspendedDiscreteUpdates !== null &&
+      componentsThatCallSuspendedDiscreteUpdates.size > 0
+    ) {
       const componentNames = [];
       componentsWithSuspendedDiscreteUpdates.forEach(name => {
         componentNames.push(name);
       });
       componentsWithSuspendedDiscreteUpdates = null;
 
-      // TODO: A more helpful version of this message could include the names of
-      // the component that were updated, not the ones that suspended. To do
-      // that we'd need to track all the components that updated during this
-      // render, perhaps using the same mechanism as `markRenderEventTime`.
+      const parentComponentNamesString = Array.from(
+        componentsThatCallSuspendedDiscreteUpdates,
+      )
+        .sort()
+        .join(', ');
+
+      const callerErroMessage = componentsThatCallSuspendedDiscreteUpdates.has(
+        HostRoot,
+      )
+        ? 'Component was suspended when root was mounted or updated'
+        : `The components that called suspense are: ${parentComponentNamesString}`;
+
+      componentsThatCallSuspendedDiscreteUpdates.clear();
```

Comment:
Set to `null` instead of clearing, same as `componentsWithSuspendedDiscreteUpdates` above.